app/lib/actions/katello/host/register.rb

```diff
@@ -67,13 +67,13 @@ def set_host_collections(system, activation_keys)
 
           host_collection_ids.each do |host_collection_id|
             host_collection = ::Katello::HostCollection.find(host_collection_id)
-            if !host_collection.unlimited_content_hosts && host_collection.max_content_hosts >= 0 &&
-               host_collection.systems.length >= host_collection.max_content_hosts
+            if !host_collection.unlimited_hosts && host_collection.max_hosts >= 0 &&
+               host_collection.systems.length >= host_collection.max_hosts
               fail _("Host collection '%{name}' exceeds maximum usage limit of '%{limit}'") %
                        {:limit => host_collection.max_content_hosts, :name => host_collection.name}
             end
           end
-          system.host_collection_ids = host_collection_ids
+          system.foreman_host.host_collection_ids = host_collection_ids
```

Comment:
right but the host object already exists in the context so lets go ahead and use that so we don't have to update this later :)